Openwrt makefile

alex-eri opened the pull request:
Place files "package/network/ipt-netflow/" in OpenWRT build and select in menuconfig.

Builds on Chaos Calmer, fixes #39 .

alex-eri commented:
Патч вынесен для того чтоб компилировать версию 2.2. Если я применяю его сразу к исходникам, то при скачивании с сорсфоржа или тага с гитхаба - для сборки будет использоваться непатченная версия. Может быть в следующей версии этот патч не понадобится.

Можете применить отсоединенный патч к исходникам (если он не ломает сборку на других системах), а после его применения поставить в таг новую версию ?

Как разберусь с опциями configure(сейчас hardcoded) и переделаю на гитхаб.

aabc commented:
То есть этот пекедж не будет собирать версию из git, а предыдущий релиз?

aabc commented:
Закоммитил это изменение Makefile в db665c04.

aabc commented:
Для релиза 2.3 надо подождать, чтоб убедиться, что последние изменения ничего не ломают, прогнать тесты как минимум.

aabc commented:
My tests passed all good with latest git head. Probably, it may be time to release v2.3.

alex-eri commented:
Added configuration menu and changed source location to github.

Помню как-то тут можно откатить коммиты и добавить в них файлы.. если хорошо разбираешься в гите, то можешь попробовать. я не осилю... а так прицепить бы этот пулреквест к версии 2,2

Когда будете делать релиз 2.3 или 2.2.1 удалите патч и поменяйте версию в мэйкфайле

aabc commented:
Я замержил ваши изменения https://github.com/aabc/ipt-netflow/commits/master
Сделайте еще, пожалуйста, README файл с пояснениями что делать - комментарии в commit log никто читать не будет, да и вообще не увидит.

aabc commented:
Почему в openwrt/Makefile `PKG_SOURCE_URL:=https://github.com/aabc/$(PKG_NAME).git`, но `URL:=http://ipt-netflow.sourceforge.net/`?

При релизе 2.3 что делать - достаточно поменять `PKG_VERSION:=2.2` на `PKG_VERSION:=2.3` и удалить файл `openwrt/patches/310-Makefile_crosscompile.patch`?

alex-eri commented:
URL ссылка на сайт проекта, ни на что не влияет. для документации там, в меню можно посмотреть инфо о пакете.

да, только надо в одном коммите всё зарелизить мне кажется. чтоб на ветке 2.2 остался с патчем, а на 2.3 уже без

aabc commented:
> чтоб на ветке 2.2 остался с патчем, а на 2.3 уже без

Что это значит? Перед релизом 2.3 я файл патча удалю как не нужный и всё или не всё? Сейчас патч есть, а на 2.3 патч накладывать смысла нет. В инструкциях в вашем Readme.md я ничего про накладывание патчей не вижу. Сказано

> Place Makefile in packages/network/ipt-netflow directory in OpenWRT bouldroot. Run make menuconfig and select package in Network/Netflow menu. Configure args partially supported.
> 
> Run make to build full firmware or make package/network/ipt-netflow/{clean,prepare,configure,compile,install} to rebuild packages.

Никакие патчи не упоминаются, значит они и не нужны. В Makefile команд накладывающих патч тоже не вижу.

alex-eri commented:
да, все правильно. так и говорю удалить перед коммитом..

там сотня мейкфайолв которые собираются в один в 1500 строк перед сборкой. патчи накладываются при условии если они там лежат рядом. я думаю не стоит заморачиваться с ними вообще если релиз скоро..

aabc commented:
I plan to release 2.3 real soon, after merging this readme update (and deleting patch).

aabc commented:
Я сделал
```
git cherry-pick \
 fba5df69d390230038232055bc2e043c9d328092 \
 6b60f6fe45d99a3a8bf65a595a7b7fb3ecd0651a \
 d53a06c81dfd87d7f0f8277354ce916ba9a0b2e5 \
 b26f39f2f4a4b0addd41387f371f4a772404af42
```
Затем `git rebase -i c230a8825` и squash всех коммитов в один. Затем `git commit --amend --date=now`. Так же убрал пробел в одной из `scp` команд после `\`.

Если вы не против всех этих изменений и если я ничего не забыл - то так все и пойдёт в релиз 2.3.

demontefacto commented:
Hello,
What about to compile this module for rampis? With above Makefile does not "work" it. Compiling end with error

alex-eri commented:
ok
